samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,126 @@ def _validate_deployment_preference_and_add_update_policy(
             )
 
 
+class SamCanary(SamResourceMacro):
+    """SAM canary macro."""
+
+    resource_type = "AWS::Serverless::Canary"
+    property_types = {
+        "FunctionName": PropertyType(False, is_str()),
+        "Handler": PropertyType(True, is_str()),
+        "Runtime": PropertyType(True, is_str()),
+        "CodeUri": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "InlineCode": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "MemorySize": PropertyType(False, is_type(int)),
+        "Tags": PropertyType(False, is_type(dict)),
+        "Tracing": PropertyType(False, is_type(bool)),
+        "Timeout": PropertyType(False, is_type(int)),
+        "Role": PropertyType(False, is_str()),
+        "Schedule": PropertyType(True, is_type(dict)),
+        "StartCanaryAfterCreation": PropertyType(True, is_type(bool)),
+        "ArtifactS3Location": PropertyType(False, one_of(is_type(dict), is_str())),
+        "FailureRetentionPeriod": PropertyType(False, is_type(int)),
+        "SuccessRetentionPeriod": PropertyType(False, is_type(int)),
+        "VpcConfig": PropertyType(False, is_type(dict)),
+        "Environment": PropertyType(False, dict_of(is_str(), is_type(dict))),
+    }
+
+    def to_cloudformation(self, **kwargs):
+        """Returns the Synthetics Canary to which this SAM Canary corresponds.
+
+        :param dict kwargs: already-converted resources that may need to be modified when converting this \
+        macro to pure CloudFormation
+        :returns: a list of vanilla CloudFormation Resources, to which this Function expands
+        :rtype: list
+        """
+        resources = []
+        synthetics_canary = self._construct_synthetics_canary()
+        resources.append(synthetics_canary)
+
+        return resources
+
+    def _construct_synthetics_canary(self):
+        """Constructs a AWS::Synthetics::Canary resource."""
+        canary = SyntheticsCanary(
+            self.logical_id, depends_on=self.depends_on, attributes=self.get_passthrough_resource_attributes()
+        )
+        canary.ArtifactS3Location = self.ArtifactS3Location
+        canary.Code = self._construct_code_dict
+        canary.ExecutionRoleArn = self.Role
+        canary.FailureRetentionPeriod = self.FailureRetentionPeriod
+        canary.Name = self.FunctionName if self.FunctionName else self.logical_id
```

Comment:
Can we put a comment about why we set canary's name to the function name?

Reply:
do you mean why we set canary name to the logical id? we set it to function name because that's the name provided by the user in the FunctionName property